### PR TITLE
Implemented: functionality to add refresh button when we do not have jobs on pipeline page (#28940kn).

### PR DESCRIPTION
**Pull request by meet-aniket**
**Screen shot:**
image

### CODE REVIEW

File: src/views/Pipeline.vue [hunk 1]
```diff
@@ -176,6 +179,7 @@
           <JobConfiguration :title="title" :job="currentJob" :status="currentJobStatus" :type="freqType" :key="currentJob"/>
         </aside>
       </main>
+      <ion-loading v-if="isLoading" show-backdrop="true" spinner="crescent" />
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:182]: We will not use the ion-loading component, instead, we will use the predefined emitter to display the loader on click and dismiss the loader once the API call is a success or failure.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue [hunk 2]
```diff
@@ -311,6 +318,18 @@ export default defineComponent({
         this.store.dispatch('job/fetchJobHistory', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => { event.target.complete() });
       }
     },
+    async reFetchJobs() {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:321]: Use same method we use for pull to refresh, do not pass event and handle case

File: src/views/Pipeline.vue (hunk 1 above)
Comment (adityasharma7) [inline src/views/Pipeline.vue:182]: I think this is not need we should add loader the same way we do for other instances

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 1 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue]: Done Sir.

File: src/views/Pipeline.vue (hunk 2 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue]: Done Sir

File: src/views/Pipeline.vue (hunk 1 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue]: Done Sir

File: src/views/Pipeline.vue [hunk 3]
```diff
@@ -262,7 +276,8 @@ export default defineComponent({
       currentJobStatus: '',
       freqType: '' as any,
       isJobDetailAnimationCompleted: false,
-      isDesktop: isPlatform('desktop')
+      isDesktop: isPlatform('desktop'),
+      isLoading: false
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:280]:
```suggestion
      isRetrying: false
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue [hunk 4]
```diff
@@ -318,12 +333,22 @@ export default defineComponent({
       })
     },
     async refreshJobs(event: any) {
+      this.isRetrying = true;
       if(this.segmentSelected === 'pending') {
-        this.store.dispatch('job/fetchPendingJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => { event.target.complete() });
+        this.store.dispatch('job/fetchPendingJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => {
+          if(event) event.target.complete();
+          this.isRetrying = false;
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:340]: We have set `isRetrying` to false, but what if the action throws some error.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 4 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue:340]: Sir, we do not return any kind of response from the action so according to me, we will always execute `.then()` statement WDYT.

**Comment by dt2patel** [comment]
@azkyakhan please update UI on this to center button properly

**Comment by azkyakhan** [comment]
> @azkyakhan please update UI on this to center button properly

I have updated UI on this to center retry button.
https://github.com/hotwax/job-manager/pull/101/commits/c1a625ae74ef8a19334d7b4608e38394f6537749

retry-button

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**